modules/library/main/src/test/java/org/geotools/feature/visitor/EqualAreaListVisitorTest.java

```diff
@@ -0,0 +1,193 @@
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.feature.visitor;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.geotools.data.DataUtilities;
+import org.geotools.data.collection.ListFeatureCollection;
+import org.geotools.factory.CommonFactoryFinder;
+import org.geotools.feature.SchemaException;
+import org.geotools.filter.function.EqualAreaFunction;
+import org.geotools.util.NumberRange;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.filter.FilterFactory2;
+import org.opengis.filter.PropertyIsBetween;
+import org.opengis.filter.expression.Expression;
+
+@RunWith(Enclosed.class)
+public class EqualAreaListVisitorTest {
+
+    static final FilterFactory2 FF = CommonFactoryFinder.getFilterFactory2();
+    public static final Expression PERSONS = FF.property("PERSONS");
+
+    /** Returns a feature collection with a simplified US geometry and PERSONS count */
+    public static ListFeatureCollection getSimplifiedStatesCollection()
+            throws SchemaException, IOException {
+        SimpleFeatureType schema =
+                DataUtilities.createType(
+                        "states", "the_geom:MultiPolygon,STATE_ABBR:String,PERSONS:Double");
+
+        try (BufferedReader br =
+                new BufferedReader(
+                        new InputStreamReader(
+                                EqualAreaListVisitorTest.class.getResourceAsStream(
+                                        "states.properties")))) {
+            List<SimpleFeature> featureList =
+                    br.lines()
+                            .map(line -> DataUtilities.createFeature(schema, line))
+                            .collect(Collectors.toList());
+            return new ListFeatureCollection(schema, featureList);
+        }
+    }
+
+    public static double getTotalArea(ListFeatureCollection fc, Expression areaFunction) {
+        return fc.stream().mapToDouble(f -> areaFunction.evaluate(f, Double.class)).sum();
+    }
+
+    /** Returns min and max value applying the expression over the feature collection */
+    public static NumberRange<Double> getMinMax(
+            ListFeatureCollection fc, Expression attributeExpression) {
+        double min =
```

Comment:
while streams are cool, does this really win over a single loop generating min and max together?

Reply:
It does not, but it's not performance critical and I find it more readable

Reply:
That said, I can turn it back to a loop, but it's just test code :-p

Reply:
Actually, found a better way to collect min and max in a single pass, will update the PR.

Reply:
I was just wondering about the speed difference really

Reply:
They are slower :-)